src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -614,6 +673,7 @@ export class SearchSource {
         // if items that are in the docvalueFields are provided, we should
         // inject the format from the computed fields if one isn't given
         const docvaluesIndex = keyBy(filteredDocvalueFields, 'field');
+        body.fields = this.getFieldsWithoutSourceFilters(index, body.fields);
```

Comment:
I haven't run this, but I'm wondering why the logic for `getFieldsWithoutSourceFilters` wouldn't be colocated with the other source filtering stuff starting on L599. They are doing very similar things in applying filters to `body.fields`, and it seems could be combined (unless there's a reason we need the "unfiltered" values between L599 and here?)

Reply:
It's been a long week for all of us, don't worry 😂  Thanks for the comments, will get address this soon.